src/main.ts

```diff
@@ -102,26 +103,17 @@ async function main() {
 function startUpPrompt(wsUrl: string, chainName: string, implName: string) {
 	const { logger } = Log;
 	const { config } = SidecarConfig;
+
 	/**
-	 * Best effort list of known public nodes that do not encourage high traffic
-	 * sidecar installations connecting to them for non - testing / development purposes.
+	 * Retrieving public endpoints from @polkadot/apps-config/endpoints
 	 */
-	const publicWsUrls: string[] = [
-		'wss://rpc.polkadot.io',
-		'wss://cc1-1.polkadot.network',
-		'wss://kusama-rpc.polkadot.io',
-		'wss://cc3-5.kusama.network',
-		'wss://fullnode.centrifuge.io',
-		'wss://crab.darwinia.network',
-		'wss://mainnet-node.dock.io',
-		'wss://mainnet1.edgewa.re',
-		'wss://rpc.kulupu.corepaper.org/ws',
-		'wss://main1.nodleprotocol.io',
-		'wss://rpc.plasmnet.io/',
-		'wss://mainnet-rpc.stafi.io',
-		'wss://rpc.subsocial.network',
-		'wss://full-nodes-lb.kilt.io:443',
-	];
+	const endpoints = createWsEndpoints(<T = string>(): T => '' as unknown as T);
+	const publicWsUrls: string[] = endpoints
+		.map(function (endpoint) {
+			if (endpoint.value && endpoint.info != 'local') return endpoint.value;
+			else return '';
```

Comment:
I had to add the `else` statement here otherwise I get the following error :
`error TS7030: Not all code paths return a value.`
So then I had some empty strings in my array which I removed by chaining a `filter` iterator after it.

Reply:
If you swap the order of `map` and `filter` I think we can get rid of that.

```ts
const publicWsUrls: string[] = endpoints
                .filter(e => e.value && e.info != 'local');
		.map(e => e.value)
```

However, I'm no `TS/JS` expert but I still think I would prefer

```ts
 const publicWsUrls: string[] = [];
 const endpoints = createWsEndpoints(<T = string>(): T => ('' as unknown as T));
 for (var endpoint of endpoints) {
         if (endpoint.value && endpoint.info != 'local') {
              publicWsUrls.push(endpoint.value);
         }
 }
```

but I have no strong opinions but let's wait for @TarikGul to take a look so I don't lie to you :)

Reply:
Yea I agree with @niklasad1 here that both these examples are improvements. I would also say i prefer the `for` loop over filter and map. 

1. As mentioned in a chat with me and @Imod7 there are performance gains (even if they are incredibly small here being that the max size of this vector would be probably no more than a 1000).

2. I think it's easier to read / debug, or modify.

Reply:
Hello @TarikGul @niklasad1 ! Thank you both for taking the time to review and your super valuable feedback! 
Two more questions : 
1. Do I need to update the changelog file or it is not necessary for this type of change ?
2. Would it be useful to add a test for this change ? eg. a test that checks that in the generated array (`publicWsUrls`) there are at least some standard public urls we always expect to have there (polkadot / kusama)?